### ISSUE TITLE
Fix example of clearing item to prevent caching (_cci1/how-cache-works.md)

### ISSUE THREAD

**Issue by lumberj** [circleci/circleci-docs#581]
This line https://circleci.com/docs/how-cache-works/#clearing-the-cache shows an example of clearing an item so it doesn't get cached:

```
dependencies:
  post:
    - rm -r ~/.gradle
```

This is incorrect, because the tests run after this command. For node projects, for example, `node_modules` gets cached. Maybe I have a dependency I'd like to always reinstall (perhaps it points to a branch on github and it should always install latest). I can't just `rm -r node_modules/my-dependency` because the tests run afterwards.

There doesn't seem to be any way to clear things from the cache and allow them to be used during the test stage.

**Comment by lumberj** [comment]
The example should probably be:

```
dependencies:
  pre:
    - rm -r ~/.gradle
```

**Comment by ryanwohara** [comment]
This is more or less fundamentally addressed in 2.0. The caching in 1.0 is less than ideal.
